### PR TITLE
JavaScript capstone project - Your API-based webapp

### PR DESCRIPTION
**Pull request by Goruchie**
# **Consuming anime API**

**The Consuming-anime-API webapp displays some characteristics of different Pokemons via the Pokemon API service. The user can share their opinions through likes and comments via the Involvement-API service.**

**When the user clicks on the "Comment" button, the data is recorded in the Involvement API, and the screen is updated with the new comment data appearing on the screen.**

[Presenting project: Video drive](https://drive.google.com/file/d/15RO6ZXV5ZMFAj4X42R2l-1Te79PZtAKo/view?usp=sharing)

**If you want to see more than 50 pokemons wait a little bit for the API response.**

## APIs used for this project:
 [Involvemenet API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270)
 [Poke API](https://pokeapi.co/)

## Counters added: 
- **Counter for likes:** 
https://github.com/J2ZROMERO/Consuming-anime-API/blob/b7f676fb06e3bc2fa24ffe97e9489176dbb1658d/src/modules/capstoneAPI.js#L36-L58
- **Counter for pokemon from API:**
https://github.com/J2ZROMERO/Consuming-anime-API/blob/b7f676fb06e3bc2fa24ffe97e9489176dbb1658d/src/modules/requestApi.js#L86-L92
- **Counters for comments:**
https://github.com/J2ZROMERO/Consuming-anime-API/blob/b7f676fb06e3bc2fa24ffe97e9489176dbb1658d/src/modules/capstoneAPI.js#L60-L81
- **Counters for a specific quantity of pokemon by default the system shows you 6 but, you can put any quantity less than 1154 and wait to receive the data:**
https://github.com/J2ZROMERO/Consuming-anime-API/blob/b7f676fb06e3bc2fa24ffe97e9489176dbb1658d/src/modules/requestApi.js#L66-L92
## Add dynamic content:
- **Window.load.** 
- **By default, the page shows you six with a specific quantity of likes (try to add more).**
- **When you click on the heart icon in a pokemon store like you did.**
- **When the pop-up appears the system shows you the pokemon with the specific data from APIs.**

## Modules:
- **Comments, pokemons, and like counters are in a unique module.**
- **Add requestApi.js module for [Poke API](https://pokeapi.co/), here are counters for likes.**
- **Add capstoneAPI.js module for [Involvemenet API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) , here are comments counters.**
- **Add popup.js module for adding popup comments and pokemon view.**
- **Add eleLiPok.js module to generate targets and comments.**

## Mocks and Tests 
- **Add a __mocks__ file and _tests_file.**
- **RequestApi.js returns a promise with a mock of a pokemon from any API and the counter for the specific quantity of pokemon.**
- **CapstoneAPI.js returns a promise with a mock of a number of comments from any API and the likes counter.**
- **Add requestApi.test.js and capstoneAPI.test.js to test the code using mock instead of real values.**

- Using gitflow, we are working in [main > dev > structure] branch.

> [Add webpack-plugin to generate its own index.html](https://webpack.js.org/guides/output-management/#setting-up-htmlwebpackplugin)

> [Add style to webpack](https://webpack.js.org/guides/asset-management/#loading-css)

> [Add local server](https://webpack.js.org/guides/development/#using-webpack-dev-server)

## Follow these steps to run the project ⚠️   
- **Step 1: execute in your terminal with the corresponding path"npm run build".**
- **Step 2: execute with the corresponding path "npm run start".**

### CODE REVIEW

File: src/modules/popup.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const addNewComment = new Requestapicapstone();
+
+const addcomment = (id) => {
+  document.querySelector('.comment-btn').addEventListener('click', () => {
+    const nameInput = document.querySelector('.nameInput').value;
+    const textInput = document.querySelector('.textInput').value;
+    addNewComment.postcoments(id, nameInput, textInput);
+    setTimeout(() => { window.location.reload(); }, 200);
+    setTimeout();
```
Comment (Omar-Muhamad) [inline src/modules/popup.js:49]: - [ ] When the user clicks on the "Comment" button, the data is recorded in the Involvement API, and the screen is updated with the new comment data appearing on the screen. Now when I click on the comment button the data is recorded in the API but the page refreshes to the home page. So please fix that.

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Goruchie and @J2ZROMERO,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:
- A description of the project.
- A demo of the project features.
- You should also highlight some interesting piece of code or something you built that you are very proud of.
- You all should appear in the video and talk about the project.
and Kindly add the link to the Pull Request summary.

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by J2ZROMERO** [comment]
**When the user clicks on the "Comment" button, the data is recorded in the Involvement API, and the screen is updated with the new comment data appearing on the screen.**

[Presenting project: Video drive](https://drive.google.com/file/d/15RO6ZXV5ZMFAj4X42R2l-1Te79PZtAKo/view?usp=sharing)

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

 - The PR title and description looks professional✔️
- Descriptive commit messages✔️
- You have a descriptive readme file✔️
- No linter errors✔️


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**